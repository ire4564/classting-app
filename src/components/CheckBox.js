import React from 'react';
import { useState } from 'react';
import ToggleButton from 'react-bootstrap/ToggleButton';
import "../App.css";

function CheckBox() {
    const [radioValue, setRadioValue] = useState('1');

    const radios = [
        { name: 'answer 1', value: '1' },
        { name: 'answer 2', value: '2' },
        { name: 'answer 3', value: '3' },
        { name: 'answer 4', value: '4' },
    ];

    return (
        <>
        {radios.map((radio, idx) => (
            <ToggleButton
                key={idx}
                id={`radio-${idx}`}
                type="radio"
                variant={idx % 2 ? 'outline-success' : 'outline-success'}
                name="radio"
                className="SelectBtn"
                value={radio.value}
                checked={radioValue === radio.value}
                onChange={(e) => setRadioValue(e.currentTarget.value)}
                >
                <div className="ProblemNum">{radio.value}</div>
                {radio.name}
            </ToggleButton>
            ))}
        </>
    );
}

export default CheckBox;