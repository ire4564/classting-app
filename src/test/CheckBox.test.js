import React from 'react';
import CheckBox from "../components/CheckBox";
import { render } from "@testing-library/react";

it("matches snapshot", () => {
    const utils = render(<CheckBox dataSet={["A", "B", "C", "D"]}/>)
    expect(utils.container).toMatchSnapshot();
});

it("shows the props correctly", () => {
    const utils = render(<CheckBox dataSet={["A", "B", "C", "D"]}/>)
    utils.getByText("A");
    utils.getByText("B");
    utils.getByText("C");
    utils.getByText("D");
})