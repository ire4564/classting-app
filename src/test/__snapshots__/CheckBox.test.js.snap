// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <label
    class="SelectBtn btn btn-outline-success"
    id="radio-0"
  >
    <input
      autocomplete="off"
      name="radio"
      type="radio"
      value="A"
    />
    <div
      class="ProblemNum"
    >
      1
    </div>
    A
  </label>
  <label
    class="SelectBtn btn btn-outline-success"
    id="radio-1"
  >
    <input
      autocomplete="off"
      name="radio"
      type="radio"
      value="B"
    />
    <div
      class="ProblemNum"
    >
      2
    </div>
    B
  </label>
  <label
    class="SelectBtn btn btn-outline-success"
    id="radio-2"
  >
    <input
      autocomplete="off"
      name="radio"
      type="radio"
      value="C"
    />
    <div
      class="ProblemNum"
    >
      3
    </div>
    C
  </label>
  <label
    class="SelectBtn btn btn-outline-success"
    id="radio-3"
  >
    <input
      autocomplete="off"
      name="radio"
      type="radio"
      value="D"
    />
    <div
      class="ProblemNum"
    >
      4
    </div>
    D
  </label>
  <div>
    <button
      class="NextBtn DisplayNone btn btn-outline-light"
      type="button"
    />
  </div>
</div>
`;
